test/uk/gov/hmrc/gform/core/TemplateValidatorSpec.scala

```diff
@@ -555,6 +547,43 @@ class TemplateValidatorSpec extends Spec {
     }
   }
 
+  it should "detect invalid references/forward references in a ValidIf Boolean expression" in {
+    val constant = Constant("")
+    val forwardRef = FormCtx(FormComponentId("fieldB"))
+    val formComponentsB = List(mkFormComponent("fieldB", Value))
+    val baseSectionB = mkSection("sectionB", formComponentsB)
+    val invalidRef = FormCtx(FormComponentId("a"))
+    val forwardReferenceError = Invalid("id 'fieldB' named in validIf is forward reference, which is not permitted")
+    val invalidReferenceError = Invalid("id 'a' named in validIf expression does not exist in a form")
+    val table =
+      Table(
+        ("booleanExpr", "expected"),
+        (Equals(forwardRef, constant), forwardReferenceError),
+        (Equals(invalidRef, constant), invalidReferenceError),
```

Comment:
Would be good to add a scenario for validIf in validators?

Reply:
Done.